library/spdm_responder_lib/libspdm_rsp_challenge_auth.c

```diff
@@ -171,8 +171,22 @@ return_status spdm_get_response_challenge_auth(IN void *context,
     spdm_get_random_number(SPDM_NONCE_SIZE, ptr);
     ptr += SPDM_NONCE_SIZE;
 
-    result = spdm_generate_measurement_summary_hash(
-        spdm_context, FALSE, spdm_request->header.param2, ptr);
+    if (spdm_is_capabilities_flag_supported(
```

Comment:
This capability check was moved out from within the spdm_generate_measurement_summary_hash in order to eliminate the need to pass the spdm_context_t.